include/flash.h

```diff
@@ -127,6 +137,28 @@ static inline int flash_write_protection_set(struct device *dev, bool enable)
 	return api->write_protection(dev, enable);
 }
 
+/**
+ *  @brief  Get layout of pages of flash.
```

Comment:
I like the run-length representation on the back-end, because it makes it possible to efficiently represent flash layouts which have many sectors of the same size, or a few sectors of variable sizes.

However, I don't think this representation is convenient for public consumption. I think that what most users will have is a starting flash offset, and they will want to know two things:

1. The base offset of the sector that flash offset is part of (or its sector index)
2. The size of thee sector that flash offset is part of

Users could implement this by consuming flash_page_layout() and writing some code on top of it, but it would be the exact same code for every target, regardless of the actual flash layout.

So I think instead that the API can require a page layout from the driver itself, but present users with generic functions (written in a flash driver core .c file) which do the arithmetic to convert between addresses and sectors/sizes.

What do you think?